### ISSUE TITLE
How can the sandboxed exe create file in c:?

### ISSUE THREAD

**Issue by tohghua** [sandboxie-plus/Sandboxie#3160]
### Describe what you noticed and did

1. Download test.exe from [https://cowtransfer.com/s/98e76eb3f5294a](url)
2. Run it to install in a sandbox, the installer will automatically run the program. See [https://cowtransfer.com/s/360bb2f7b83348](url)
3. Input anything such as “a” and press Ok button, it will create c:\CarpaBrowserCache

The problem is why the program can break through the sandbox to create folder directly in disk.

### Affected program

test.exe

### Download link

https://cowtransfer.com/s/98e76eb3f5294a

### Where is the program located?

The program is installed only inside a sandbox (NOT in the real system anyway).

### Expected behavior

It should unable to create file/folder in c:

### What is your Windows edition and version?

win10

### In which Windows account you have this problem?

A local account (Administrator).

### Please mention any installed security software

none

### What version of Sandboxie are you running?

Sandboxie Plus 1.9.8

### Is it a new installation of Sandboxie?

I recently did a new clean installation.

### In which sandbox type you have this problem?

In a standard isolation sandbox (yellow sandbox icon).

### Can you reproduce this problem on a new empty sandbox?

I can confirm it also on a new empty sandbox.

**Comment by DavidXanatos** [comment]
I cant reproduce this anyone else being able to reproduce this issue?

**Comment by coltsfootrock** [comment]
Not really too sure what I did to reproduce it as the exe was not in English my native language. Just downloaded it and ran in in yellow Sbie+ box and clicked on the highlighted boxes during the install.
Used a yellow box
![Screenshot 2023-08-04 180330](https://github.com/sandboxie-plus/Sandboxie/assets/135013941/b2e27b62-9d62-483d-bdde-da49b85a70fb)

**Comment by bastik-1001** [comment]
I am also not able to reproduce this. On Windows 7, the CarpaBrowser crashes, after being installed, no files are created outside the sandbox. On another machine that runs Windows 10, the installer is able to launch the browser, but when I type a (or something else) and click the left-most button, a message appears telling me that it failed to write to the drive with the letter d.

There is just C and it is not a "real" computer, just a device running Windows 10.

**Comment by offhub** [comment]
https://github.com/sandboxie-plus/Sandboxie/assets/6871698/f5691b3e-2f3a-4ac9-8da1-bc94704e9401

**Comment by offhub** [comment]
`ClosedClsid={00021401-0000-0000-C000-000000000046}`

**Comment by bastik-1001** [comment]
It appears as if messing with the HKCU does not require a process to be elevated, whereas it needs to be for messing with HKLM?

Does this affect the boxes where the processes are not allowed to be started elevated? And what about security hardened boxes?

**Comment by DavidXanatos** [comment]
the registry changes should be contained within the box, did you observed otherwise?
in case of this specific browser bug it seams it arises when allowing to communicate with a unsandboxed windows component
try removing Template=OpenWinInetCache from [DefaultTemplates] in the templates.ini

**Comment by bastik-1001** [comment]
> did you observed otherwise?

No, I did not. My comment was based on the `ClosedClsid={00021401-0000-0000-C000-000000000046}` being suggested as a possible workaround.

**Comment by offhub** [comment]
This can also be used as a workaround.

`NormalIpcPath=\RPC Control\webcache_*`

**Comment by isaak654** [comment]
Considering the OpenWinInetCache template:

https://github.com/sandboxie-plus/Sandboxie/blob/169344d5ae410e5d912b4abbe0483ff13d4ec717/Sandboxie/install/Templates.ini#L462-L470

 - 1 - Is there a way to configure it so that it is only applied under certain conditions (such as the presence or absence of specific registry keys/files/folders)?

 - 2.1 - What about replacing line 468 with `OpenIpcPath=!executable.exe,\RPC Control\webcache_*` ?

  - 2.2 - What about using a similar approach for line 645 below?
https://github.com/sandboxie-plus/Sandboxie/blob/169344d5ae410e5d912b4abbe0483ff13d4ec717/Sandboxie/install/Templates.ini#L639-L647
